src/main/java/bio/terra/common/iam/AuthenticatedUserRequest.java

```diff
@@ -0,0 +1,110 @@
+package bio.terra.common.iam;
+
+import bio.terra.common.exception.ApiException;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import java.util.Objects;
+import java.util.UUID;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+
+@JsonDeserialize(builder = AuthenticatedUserRequest.Builder.class)
+public class AuthenticatedUserRequest {
```

Comment:
nit: all public classes should have a javadoc comment that's hopefully enlightening.
https://google.github.io/styleguide/javaguide.html#s7.3-javadoc-where-required